### PR TITLE
change master to main references

### PR DESCRIPTION
**Pull request by bjlittle**
### Description
After the renaming of the `iris/master` branch to `iris/main`, this pull-request changes all relevant `master` references to be `main` within `iris`.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

**Comment by bjlittle** [comment]
`isort` is misbehaving under `nox` with `pip`... I'll take a look at that :+1:

File: .cirrus.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ linux_task_template: &LINUX_TASK_TEMPLATE
 # YAML alias for compute credits.
 #
 compute_credits_template: &CREDITS_TEMPLATE
-  # Only use credits for non-DRAFT pull-requests to SciTools/iris master branch by collaborators
+  # Only use credits for non-DRAFT pull-requests to SciTools/iris main branch by collaborators
```
Comment (trexfeathers) [inline .cirrus.yml:90]: [From what I understand about Cirrus](https://cirrus-ci.org/guide/writing-tasks/), `use_compute_credits` will also be true for merges onto feature branches. To achieve what is stated in the comment, we would also need a comparison between `CIRRUS_BASE_BRANCH` and `CIRRUS_DEFAULT_BRANCH`.

File: README.md [hunk 2]
```diff
@@ -11,13 +11,13 @@
 
 <p align="center">
 <a href="https://cirrus-ci.com/github/SciTools/iris">
-<img src="https://api.cirrus-ci.com/github/SciTools/iris.svg?branch=master"
+<img src="https://api.cirrus-ci.com/github/SciTools/iris.svg?branch=main"
```
Comment (trexfeathers) [inline README.md:14]:
```suggestion
<img src="https://api.cirrus-ci.com/github/SciTools/iris.svg"
```

Still works, simpler.

File: docs/src/whatsnew/2.3.rst [hunk 3]
```diff
@@ -238,7 +238,7 @@ Documentation
 =============
 
 * Adopted a
-  `new colour logo for Iris <https://github.com/SciTools/iris/blob/master/docs/src/_static/Iris7_1_trim_100.png>`_
+  `new colour logo for Iris <https://github.com/SciTools/iris/blob/main/docs/src/_static/Iris7_1_trim_100.png>`_
```
Comment (trexfeathers) [inline docs/src/whatsnew/2.3.rst:241]: 😉

File: docs/src/developers_guide/gitwash/development_workflow.rst [hunk 4]
```diff
@@ -37,7 +37,7 @@ See `linux git workflow`_ for some explanation.
 Consider Deleting Your Master Branch
```
Comment (trexfeathers) [inline docs/src/developers_guide/gitwash/development_workflow.rst:37]:
```suggestion
Consider Deleting Your Main Branch
```

For consistency.

File: docs/src/whatsnew/latest.rst [hunk 5]
```diff
@@ -147,7 +147,7 @@ This document explains the changes made to Iris for this release
    environment info. (:pull:`3990`)
 
 #. `@bjlittle`_ enabled `cirrus-ci`_ compute credits for non-draft pull-requests
-   from collaborators targeting the Iris ``master`` branch. (:pull:`4007`)
+   from collaborators targeting the Iris ``main`` branch. (:pull:`4007`)
```
Comment (trexfeathers) [inline docs/src/whatsnew/latest.rst:150]: If we include entries like this in What's New (which I approve of), then this PR itself deserves an entry too.

**Review by trexfeathers — CHANGES_REQUESTED** [review]
Good stuff @bjlittle ❤

See my comments below.

Beyond that, I can confirm that all remaining mentions of the word "master" in the repo are unrelated to Iris' own branch structure.

**Comment by trexfeathers** [comment]
>`isort` is misbehaving under `nox` with `pip`... I'll take a look at that :+1: 

We've suffered due to isort's latest version changing under our feet. The intent to make pre-commit's versions king - discussed in #4181 - would avoid this happening in future. Either:

* @bjlittle's preferred solution of scraping the versions from the pre-commit file before installing the lint packages
  OR
* [The alternative that I have recently proposed for cf-units that just uses pre-commit for all linting.](https://github.com/SciTools/cf-units/blob/134d03dfb75a257ba4588c5809e016c07266563b/.cirrus.yml#L27-L39)

File: noxfile.py [hunk 6]
```diff
@@ -172,9 +175,10 @@ def flake8(session: nox.sessions.Session):
     # Pip install the session requirements.
     session.install("flake8")
     # Execute the flake8 linter on the package.
-    session.run("flake8", PACKAGE)
-    # Execute the flake8 linter on this file.
-    session.run("flake8", __file__)
+    session.run("flake8", str(PACKAGE))
```
Comment (trexfeathers) [inline noxfile.py:178]: `PACKAGE` is already a `str`. What's up with wrapping it again?

File: README.md (hunk 2 above)
Comment (bjlittle) [reply README.md:14]: There is no need to change this.

File: docs/src/whatsnew/latest.rst (hunk 5 above)
Comment (bjlittle) [reply docs/src/whatsnew/latest.rst:150]: Added a whatsnew entry

File: noxfile.py (hunk 6 above)
Comment (bjlittle) [reply noxfile.py:178]: Yup, I refactored and pushed that up

**Review by trexfeathers — APPROVED** [review]

**Comment by bjlittle** [comment]
@trexfeathers Awesome, thanks 👍

**Pull request merged**